### PR TITLE
Support `colspan` attribute in `table` HTML, including when pasting

### PR DESCRIPTION
**Pull request by mpkelly**
## What?
Support `colspan` attribute on `th` and `td` when pasting tables in the Editor. 

## Why?
It was based on [this](https://github.com/WordPress/gutenberg/issues/45774#issuecomment-1316052739) requirement in https://github.com/WordPress/gutenberg/issues/45774

## Testing Instructions

1. Check out this branch and try pasting tables in from external sources. The `colspan` attribute should be supported on `th` and `td` elements within `thead`, `tbody` and `tfoot` elements. 

## Screenshots or screencast 

image

### CODE REVIEW

**Review by danielbachhuber — COMMENTED** [review]
Thanks for the PR, @mpkelly !

The table seems to be formatted correctly on paste:

image

However, I don't see any `colspan=""` on the serialized data...

image

Then, when I refresh, the formatting is lost:

image

Maybe `packages/block-library/src/table/save.js` needs to handle `colspan`? It'd probably be good to have update the tests for this particular scenario, too.

**Comment by mpkelly** [comment]
Thanks for the review, @danielbachhuber. 

I will try and get this working now. I would also like to look at the list transformation you mentioned too.

**Comment by mpkelly** [comment]
It's now saving for me, and I can view it in a preview. I tried to add tests - just some snapshot tests - but it was too difficult. These simple components implicitly reference all kinds of stuff, which is likely why we don't have tests for the same functions in other blocks.

**Review by danielbachhuber — APPROVED** [review]
Works great in my testing!

The `<thead>` and `<tfoot>` persist through save:

image

File: packages/block-library/src/table/transforms.js [hunk 1]
```diff
@@ -1,15 +1,23 @@
+/**
+ * WordPress dependencies
+ */
+/**
+ * Internal dependencies
+ */
```
Comment (ellatrix) [inline packages/block-library/src/table/transforms.js]: Minor: these should be removed?

Comment (danielbachhuber) [reply packages/block-library/src/table/transforms.js]: Sure! Removed in 041661de35

File: packages/blocks/src/api/raw-handling/test/paste-handler.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                <td colspan="2"
+                    style="border-left:solid #000000 1pt;border-right:solid #000000 1pt;border-bottom:solid #000000 1pt;border-top:solid #000000 1pt;vertical-align:top;padding:5pt 5pt 5pt 5pt;overflow:hidden;overflow-wrap:break-word;">
+                    <p dir="ltr" style="line-height:1.2;margin-top:0pt;margin-bottom:0pt;"><span
+                            style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Test colspan</span>
+                    </p></td>
+            </tr>
+            </tbody>
+        </table>
+    </div>
+    <br/><br/></b>
```
Comment (ellatrix) [inline packages/blocks/src/api/raw-handling/test/paste-handler.js:51]: Minor: this maybe makes more sense to add to the [integration tests](https://github.com/WordPress/gutenberg/blob/trunk/test/integration/blocks-raw-handling.test.js), which has all blocks loaded.

**Review by ellatrix — APPROVED** [review]
Looks good to me! Will be interesting to add some UI for this. 😄

**Comment by mpkelly** [comment]
Thanks for the reviews and merge! I will look at moving the test in another PR.

File: packages/blocks/src/api/raw-handling/test/paste-handler.js (hunk 2 above)
Comment (danielbachhuber) [reply packages/blocks/src/api/raw-handling/test/paste-handler.js:51]: @ellatrix Would it make sense to keep this existing test _and_ add it to integration tests?

**Pull request merged**